File: openmrs-client/src/main/java/org/openmrs/mobile/activities/providerdashboard/ProviderDashboardPagerAdapter.java
package org.openmrs.mobile.activities.providerdashboard;

import android.content.Context;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;


import java.util.ArrayList;
import java.util.List;

public class ProviderDashboardPagerAdapter extends FragmentPagerAdapter {

    private static final int TAB_COUNT = 2;

    private final List<Fragment> mFragmentList = new ArrayList<>();
    private final List<String> mFragmentTitleList = new ArrayList<>();

    private Context context;

    ProviderDashboardPagerAdapter(FragmentManager fm, Context context) {
        super(fm);
        this.context = context;
    }

    @Override
    public Fragment getItem(int i) {
        return mFragmentList.get(i);
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return mFragmentTitleList.get(position);
    }

    public void addFragment(Fragment fragment, String title) {
        mFragmentList.add(fragment);
        mFragmentTitleList.add(title);
    }

    @Override
    public int getCount() {
        return TAB_COUNT;
    }

}

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/providerdashboard/ProviderDashboardPagerAdapter.kt
package org.openmrs.mobile.activities.providerdashboard

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import java.util.ArrayList

class ProviderDashboardPagerAdapter constructor(fm: FragmentManager?) : FragmentPagerAdapter(fm!!) {
    private val mFragmentList: MutableList<Fragment> = ArrayList()
    private val mFragmentTitleList: MutableList<String> = ArrayList()

    fun addFragment(fragment: Fragment, title: String) {
        mFragmentList.add(fragment)
        mFragmentTitleList.add(title)
    }

    override fun getItem(i: Int): Fragment = mFragmentList[i]

    override fun getPageTitle(position: Int): CharSequence = mFragmentTitleList[position]


    override fun getCount(): Int = TAB_COUNT

    companion object {
        private const val TAB_COUNT = 2
    }
}
